scripts/amoebagame.js

```diff
@@ -1,40 +1,70 @@
+        poke = [
+            {species: "Solosis", cry: "solosis solosis", icon: 577},
+            {species: "Lilligant", cry: "lilli-lilligant", icon: 549},
+            {species: "Krokorok", cry: "krokokokokorok!", icon: 552},
+            {species: "Minccino", cry: "minccino?", icon: 572},
+            {species: "Mankey", cry: "ma-mankey mankey!", icon: 56},
+            {species: "Arceus", cry: "-", icon: 493}
+        ],
+        battles = {},
+        htmlEscape = utilities.html_escape;
 
-    var init = function() {
+    function pokeIcon(num) {
+        return "<img src='icon:" + poke[num].icon + "'>";
+    }
+
+    this.init = function () {
         if (sys.existChannel(name)) {
             id = sys.channelId(name);
         } else {
             id = sys.createChannel(name);
         }
         SESSION.global().channelManager.restoreSettings(id);
+        if (!SESSION.channels(id).topic) {
+            SESSION.channels(id).topic = defaultTopic;
+        }
         SESSION.channels(id).perm = true;
         SESSION.channels(id).master = "lamperi";
     };
-
-    var beforeChatMessage = function(src, message, chan) {
-        if (chan != id) return false;
-        var role = SESSION.users(src).amoeba.role;
-        if (sys.auth(src) > 0 && ["/","!"].indexOf(message[0]) > -1) return false;
-        if (role != species.length - 1) {
-            sendChanAll(sys.name(src) + ": " + messages[role], id);
-            if (battles[role] && battles[role] != src && sys.isInChannel(battles[role], id)) {
-                var winner = sys.rand(0,2) > 0 ? battles[role] : src;
-                var loser = winner == src ? battles[role] : src;
-                SESSION.users(winner).amoeba.role++;
-                if (SESSION.users(loser).amoeba.role > 0) SESSION.users(loser).amoeba.role--;
-                sendChanAll("+EvolutionBot: Two " + species[role] + "s, "
-                    + sys.name(battles[role]) + " and " + sys.name(src) + " engaged in a fierce battle! "
-                    + sys.name(winner) + " won and is now a " + species[SESSION.users(winner).amoeba.role] + ". "
-                    + sys.name(loser) + " lost and is now a " + species[SESSION.users(loser).amoeba.role] + ".",
-                id);
+    this.beforeChatMessage = function (src, message, channel) {
+        if (channel !== id) {
+            return false;
+        }
+        var role = SESSION.users(src).amoeba.role,
+            isAuth = sys.auth(src) > 0 && sys.auth(src) < 4, // ignores hidden auth
+            name = htmlEscape(sys.name(src)),
+            messagePrefix = "<font color='" + script.getColor(src) + "'><timestamp/>" + pokeIcon(role) + (isAuth ? "+" : "") + "<b>" + (isAuth ? "<i>" + name + ":</i> " : name + ": ") + "</b></font>";

```

Comment:
The icon seems redundant. I mean, you don't see it unless they talk. When they talk, they say what role they are. This just needlessly spaces out the chat.